### ISSUE TITLE
[💡 Feature]: Allow custom commands scoped to element to be chainable in async mode

### ISSUE THREAD

**Issue by nextlevelbeard** [webdriverio/webdriverio#8043]
### Is your feature request related to a problem?

With async mode, we can do this:
```js
await $("id=foo").$("bar").click();
```
but if we add a custom command to the element scope that we would like to chain:
```js
await browser.addCommand('$foo', async function () {
  // Do something with an element
  return element;
}, true)
```
and try to do this:
```js
await $("id=asd").$foo("bar").click();
```
we get a  `TypeError: $(...).$foo(...).click is not a function`

Context:
I'm trying to add a custom locator strategy through the `addCommand` feature.

### Describe the solution you'd like.

What I'm requesting is that if:
- The custom command is for the element scope
- The custom command returns an element (promise)

Then that command should be chainable in `async` mode.

### Describe alternatives you've considered.

I've considered using the custom locator strategy feature: `custom$`.
I cannot use it for this purpose because that only works in a web context (not my use case).

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by christian-bromann** [comment]
Thanks for reporting!

I am totally for this and I believe it is just a matter on how these commands are being wrapped.

Any contributions that resolves the bug are highly appreciated. Please take a look into our [contribution guidelines](https://webdriver.io/docs/contribute) and let us know if you have any questions. Cheers!

**Comment by nextlevelbeard** [comment]
Do you think you can point me in the right direction?
What might be an workaround for someone dealing with this problem?
What wrapper function would I import and how could I wrap an element result at runtime?

Something like
```js
import { wrapper } from 'webdriverio/???'
await browser.addCommand('$foo', async function () {
  // Do something with an element
  return wrapper(element);
}, true)
```
or
```js
import { wrapper } from 'webdriverio/???'
await browser.addCommand('$foo', wrapper(async function () {
  // Do something with an element
  return element;
}), true)
```

**Comment by christian-bromann** [comment]
> What might be an workaround for someone dealing with this problem?

I honestly haven't invested time looking into a workaround because I would be more interested into a fix.

> Do you think you can point me in the right direction?

I think we should investigate where the core [command wrapper](https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-utils/src/shim.ts#L166-L345) fails to properly chain the commands.

**Comment by jayrepo** [comment]
Just ran into the same problem.
It's caused by the command wrapper only chain commands in the `ELEMENT_QUERY_COMMANDS `
https://github.com/webdriverio/webdriverio/blob/92f073f8d54b51ff7473d2318fcec9927608c7ca/packages/wdio-utils/src/shim.ts#L70-L73

An effortless way to fix it will be assuming the user added commands which need to be chained contains `$`.
Or we need to add a config which allows users to add custom command to the list.

**Comment by christian-bromann** [comment]
@jayrepo I think this is a good solution you are suggesting. We would chain commands as long as they end with `$` or `$$`.

**Comment by erwinheitzman** [comment]
Resolved by #8134
